pkgs/html/test/parser_feature_test.dart

```diff
@@ -265,11 +265,11 @@ On line 4, column 3 of ParseError: Unexpected DOCTYPE. Ignored.
   });
 
   test('Element.text', () {
-    final doc = parseFragment('<div>foo<div>bar</div>baz</div>');
+    final doc = parseFragment('<div>foo<div><br>bar</div>baz<br></div>');
     final e = doc.firstChild!;
     final text = e.firstChild!;
     expect((text as Text).data, 'foo');
-    expect(e.text, 'foobarbaz');
+    expect(e.text, 'foo\nbarbaz\n');
```

Comment:
I wouldn't change the behavior of `.text`, in the link you shared we had a function with an explicit boolean like `textContent({bool convertBRsToNewlines = false})`, I'd prefer to add that instead.